### PR TITLE
These Changes are Implemented by Haseena

### PR DESCRIPTION
**Pull request by RajeevKumar-Singh**

### CODE REVIEW

File: src/main/java/com/techsophy/tsf/wrapperservice/service/impl/ProcessServiceImpl.java [hunk 1]
```diff
@@ -413,6 +411,50 @@ public void completeTaskWithChecklistItemId(ChecklistItemInstanceDTO checklistIt
...
+        HttpHeaders deletehttpHeaders = new HttpHeaders();
+        deletehttpHeaders.add(HttpHeaders.AUTHORIZATION, getBearerToken());
+        deletehttpHeaders.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<?> deletehttpEntity = new HttpEntity<Object>(null, deletehttpHeaders);
+        ResponseEntity<Object> response = restTemplate.exchange(uriComponentsBuilder2.toUriString(), HttpMethod.DELETE, deletehttpEntity, Object.class);
+
+        if(response.getStatusCode().is2xxSuccessful())
+        {
+            StatusCodeUpdateDTO statusCodeUpdateDTO = new StatusCodeUpdateDTO();
+            statusCodeUpdateDTO.setFormId("951380543294529536");
```
Comment (mraviraja) [inline src/main/java/com/techsophy/tsf/wrapperservice/service/impl/ProcessServiceImpl.java:440]: Why hardcoded id?

File: src/main/java/com/techsophy/tsf/wrapperservice/service/impl/ProcessServiceImpl.java [hunk 2]
```diff
@@ -413,6 +411,50 @@ public void completeTaskWithChecklistItemId(ChecklistItemInstanceDTO checklistIt
...
+                    .writeValueAsString(statusCodeUpdateDTO);
+            String updateUrl=gatewayURI+FORM_DATA_UPDATE_URL;
+            UriComponentsBuilder uriBuilder = UriComponentsBuilder.fromHttpUrl(updateUrl);
+            HttpHeaders httpHeaders = new HttpHeaders();
+            httpHeaders.add(HttpHeaders.AUTHORIZATION, getBearerToken());
+            httpHeaders.setContentType(MediaType.APPLICATION_JSON);
+            HttpEntity<?> httpEntity = new HttpEntity<Object>(apiData,httpHeaders);
+            ResponseEntity responseEntity = restTemplate.exchange(uriBuilder.toUriString(), HttpMethod.PATCH, httpEntity, Object.class);
+
+        }
```
Comment (mraviraja) [inline src/main/java/com/techsophy/tsf/wrapperservice/service/impl/ProcessServiceImpl.java:455]: Need to handle failure response as well

File: src/main/java/com/techsophy/tsf/wrapperservice/service/impl/ProcessServiceImpl.java (hunk 1 above)
Comment (haseena-sk) [reply src/main/java/com/techsophy/tsf/wrapperservice/service/impl/ProcessServiceImpl.java:440]: Here, we are using same form for storing the data each time.

File: src/main/java/com/techsophy/tsf/wrapperservice/service/impl/ProcessServiceImpl.java (hunk 2 above)
Comment (haseena-sk) [reply src/main/java/com/techsophy/tsf/wrapperservice/service/impl/ProcessServiceImpl.java:455]: okay, i will modify

**Pull request closed without merging**